### PR TITLE
Bump CAPI to v1.4.4

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind support

**What this PR does / why we need it**:
This PR bumps CAPI to v1.4.4

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

**Release note**:

```release-note
Bump CAPI to v1.4.4
```

### CODE REVIEW

**Comment by Ankitasw** [comment]
Not sure what is wrong with the E2E tests, it's not able to create kind cluster

**Comment by damdo** [comment]
The capi migration guide for provider states that the `kind` version for providers willing to migrate to `v1.4.x` is `v0.18.x`: https://github.com/kubernetes-sigs/cluster-api/blob/4d76266bf350edcd3e5dffd18edc3690784f122d/docs/book/src/developer/providers/migrations/v1.3-to-v1.4.md#dependencies

So that might be why. Should we keep `sigs.k8s.io/kind` at `v0.18.0` and see if that works?

**Comment by damdo** [comment]
@Ankitasw @richardcase The capi migration guide for provider states that the `kind` version for providers willing to migrate to `v1.4.x` is `v0.18.x`: https://github.com/kubernetes-sigs/cluster-api/blob/4d76266bf350edcd3e5dffd18edc3690784f122d/docs/book/src/developer/providers/migrations/v1.3-to-v1.4.md#dependencies

So that might be why. Should we keep `sigs.k8s.io/kind` at `v0.18.0` and see if that works?

**Comment by richardcase** [comment]
Thanks @damdo . I think @Ankitasw is out, so i will revert the kind version and give it a go.

**Comment by Ankitasw** [comment]
@richardcase @damdo the above migration guide is to move fRom CAPI 1.3.x to 1.4.x, but this branch is already using CAPI v1.4.x.

**Comment by Ankitasw** [comment]
There are some breaking changes in kind [v0.20.0](https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0) which might be causing the issue, and which has to be fixed.

**Comment by Ankitasw** [comment]
I ll revert kind to v0.18

**Comment by richardcase** [comment]
Ha, I made a similar change yesterday with reverting kind to 0.18 and thought I'd pushed it to your branch before triggering the tests again yesterday. But looks like i didn't push it, doh.

Thanks @Ankitasw for following up on this. Lets see how the tests go, if it still fails we can do the v2.2.0 release without it.

**Comment by Ankitasw** [comment]
@richardcase same error 😞 
You can go ahead with the release I suppose, this might take time, as CAPZ is also facing similar issue.

**Comment by damdo** [comment]
I saw your changes on this branch @richardcase but I think they might've been overridden by @Ankitasw 's push.
Though with the ones you proposed @richardcase saw both the standard e2e and the e2e-eks passing: see [1](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/4372/pull-cluster-api-provider-aws-e2e/1676644773332520960) and [2](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/4372/pull-cluster-api-provider-aws-e2e-eks/1676679384599105536).

Should we bring back that branch state?

**Comment by Ankitasw** [comment]
@damdo I don't see any changes pushed by @richardcase

File: go.mod [hunk 1]
```diff
@@ -159,7 +159,7 @@ require (
 	k8s.io/kube-openapi v0.0.0-20230109183929-3758b55a6596 // indirect
 	k8s.io/kubectl v0.26.0 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
-	sigs.k8s.io/kind v0.18.0 // indirect
+	sigs.k8s.io/kind v0.20.0 // indirect
```
Comment (Ankitasw) [inline go.mod:162]: @richardcase I guess the problem is, this will always point to v0.20.0 as its indirect import from CAPI 1.4.4

Comment (Ankitasw) [reply go.mod:165]: I have made the changes to replace this I think now the tests should pass.

Comment (Ankitasw) [reply go.mod:165]: Working now 🙂

Comment (Ankitasw) [reply go.mod:165]: Thanks @damdo for pointing to the right cause 🎉

Comment (damdo) [reply go.mod:165]: Yes, that's why Richard did the pinning in the root go.mod in the first place, I think.
I saw you now did it as well, so let's see if the e2e pass this time :)

**Comment by damdo** [comment]
@Ankitasw this was the commit that @richardcase pushed https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/1f9f2dbb56524e181d373ff912de7b45262eb145 which they got overridden by a force push, which is basically what you also just did with https://github.com/kubernetes-sigs/cluster-api-provider-aws/compare/ee7028e1642c6f680c8655a820742ce2ce0766e6..7e61ad4adc76286aad6b8d9d6838ae6a06d15c9e

So I'm expecting that should be passing considering I saw it passing also previously.

File: go.mod (hunk 1 above)
Comment (richardcase) [reply go.mod:165]: :crossed_fingers:

**Comment by richardcase** [comment]
> @Ankitasw this was the commit that @richardcase pushed [1f9f2db](https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/1f9f2dbb56524e181d373ff912de7b45262eb145) which they got overridden by a force push, which is basically what you also just did with https://github.com/kubernetes-sigs/cluster-api-provider-aws/compare/ee7028e1642c6f680c8655a820742ce2ce0766e6..7e61ad4adc76286aad6b8d9d6838ae6a06d15c9e
> 
> So I'm expecting that should be passing considering I saw it passing also previously.

That's good I did push it, I had a moment of doubt this morning that i'd forgotten :rofl:

File: go.mod [hunk 2]
```diff
@@ -2,7 +2,10 @@ module sigs.k8s.io/cluster-api-provider-aws/v2
 
 go 1.19
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.4.3
+replace (
+	sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.4.4
+	sigs.k8s.io/kind => sigs.k8s.io/kind v0.18.0
```
Comment (richardcase) [inline go.mod:7]: Is it worth doing the same in hack/tools?

Comment (Ankitasw) [reply go.mod:7]: hack/tools already have direct import of kind, so it would take affect, I have changed it to v0.18.0 already.

**Comment by damdo** [comment]
It looks like the eks e2e didn't trigger? 🤔

**Comment by Ankitasw** [comment]
> It looks like the eks e2e didn't trigger? 🤔

It's in the queue to run

**Comment by richardcase** [comment]
Its got past the point of the failure :tada:

**Comment by Ankitasw** [comment]
@richardcase @damdo Looks like migration guide has been updated and required us to use kind v0.20.0

File: hack/ensure-kind.sh [hunk 3]
```diff
@@ -19,7 +19,7 @@ set -o nounset
 set -o pipefail
 
 GOPATH_BIN="$(go env GOPATH)/bin/"
-MINIMUM_KIND_VERSION=v0.14.0
+MINIMUM_KIND_VERSION=v0.18.0
```
Comment (richardcase) [inline hack/ensure-kind.sh:22]: Shall we change this to `v0.20.0`?

**Comment by richardcase** [comment]
The e2e passed:

- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/4372/pull-cluster-api-provider-aws-e2e-eks/1677211089118433280
- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/4372/pull-cluster-api-provider-aws-e2e/1677238009407213568

File: hack/ensure-kind.sh (hunk 3 above)
Comment (Ankitasw) [reply hack/ensure-kind.sh]: done

**Comment by richardcase** [comment]
As the e2e passed  i'd say we don't need to rune the e2e agai, so:

/lgtm
/approve

**Comment by damdo** [comment]
Thanks @Ankitasw 
LGTM

**Pull request merged**